pkg/volume/glusterfs/glusterfs.go

```diff
@@ -1187,6 +1187,14 @@ func (plugin *glusterfsPlugin) ExpandVolumeDevice(spec *volume.Spec, newSize res
 	if err != nil {
 		return oldSize, fmt.Errorf("failed to get volumeID for volume %s: %v", volumeName, err)
 	}
+
+	if volumeID == volumeName {
+		// If the volume was provisioned/created statically
```

Comment:
The comment and the error message is misleading. Even a "static PV", i.e. a PV created by cluster admin, can have valid volume ID and be resizeable.

IMO you should check if the volume has enough info to resize it and report what field / annotation is missing.

Reply:
@jsafrane If the volume is statically provisioned, we miss the annotation in pv spec and its hard to judge how admin provisioned the volume and from which server. Thats why we are skipping the resize on it.

Reply:
Again, even a "static PV" can have valid volume ID annotation and be resizeable. Admin can create such PV easily.

You should report that "PV misses annotation XYZ and cannot be resized". And the comment should reflect it.